### PR TITLE
Provide the proper language to the mailer

### PR DESCRIPTION
**Pull request by rullzer**
Else we can't properly translate the footer in the recipients e-mail language.

1. Create 2 users (userA and userB)
2. set language of userB to something other than English
3. share a file from userA to user B

before:
* footer in language of userA

After:
* footer in language of userB

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Of course :see_no_evil:

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -890,9 +890,9 @@ protected function sendMailNotification(IL10N $l,
 		$initiatorEmail = $initiatorUser->getEMailAddress();
 		if($initiatorEmail !== null) {
 			$message->setReplyTo([$initiatorEmail => $initiatorDisplayName]);
-			$emailTemplate->addFooter($instanceName . ($this->defaults->getSlogan() !== '' ? ' - ' . $this->defaults->getSlogan() : ''));
+			$emailTemplate->addFooter($instanceName . ($this->defaults->getSlogan($l->getLanguageCode()) !== '' ? ' - ' . $this->defaults->getSlogan($l->getLanguageCode()) : ''));
 		} else {
-			$emailTemplate->addFooter();
+			$emailTemplate->addFooter('', $l->getLanguageCode());
```
Comment (nickvergessen) [inline lib/private/Share20/Manager.php]: debugging? addFooter has only 1 parameter

File: lib/private/legacy/util.php [hunk 2]
```diff
@@ -428,6 +442,8 @@ public static function copySkeleton($userId, \OCP\Files\Folder $userDirectory) {
 			throw new \RuntimeException('username is reserved name: ' . $appdata);
 		}
 
+		$logger->debug('Final skeleton folder:' . $skeletonDirectory, ['app' => 'files_skeleton']);
```
Comment (nickvergessen) [inline lib/private/legacy/util.php]: :( so many logs for every user?

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Looks good in general, but the addFooter call should be fixed

File: lib/private/legacy/util.php (hunk 2 above)
Comment (rullzer) [reply lib/private/legacy/util.php]: Ah shit no. this was for smething else

File: lib/private/Share20/Manager.php (hunk 1 above)
Comment (rullzer) [reply lib/private/Share20/Manager.php]: No I added this. I need to know the language code.

Comment (rullzer) [reply lib/private/Share20/Manager.php]: ah let me fix the interface

**Review by juliusknorr — APPROVED** [review]

**Review by georgehrke — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
> There were 26 errors:
> ---
> 1) Test\Mail\EMailTemplateTest::testEMailTemplateCustomFooter
>
> TypeError: Argument 3 passed to OC\Mail\EMailTemplate::__construct() must implement interface OCP\L10N\IFactory, instance of Mock_IL10N_58c1d8ad given, called in /drone/src/tests/lib/Mail/EMailTemplateTest.php on line 54

**Comment by nickvergessen** [comment]
>  Lint / php-cs check (pull_request) 

failing

File: apps/theming/lib/ThemingDefaults.php [hunk 3]
```diff
@@ -145,8 +145,8 @@ public function getBaseUrl() {
 		return $this->config->getAppValue('theming', 'url', $this->url);
 	}
 
-	public function getSlogan() {
-		return \OCP\Util::sanitizeHTML($this->config->getAppValue('theming', 'slogan', parent::getSlogan()));
+	public function getSlogan(string $lang = null) {
```
Comment (nickvergessen) [inline apps/theming/lib/ThemingDefaults.php:148]:
```suggestion
	public function getSlogan(?string $lang = null) {
```

File: lib/public/Defaults.php [hunk 4]
```diff
@@ -137,8 +137,8 @@ public function getEntity() {
 	 * @return string
 	 * @since 6.0.0
 	 */
-	public function getSlogan() {
-		return $this->defaults->getSlogan();
+	public function getSlogan(string $lang = null) {
```
Comment (nickvergessen) [inline lib/public/Defaults.php:140]:
```suggestion
	public function getSlogan(?string $lang = null) {
```

File: lib/public/Mail/IEMailTemplate.php [hunk 5]
```diff
@@ -140,10 +140,11 @@ public function addBodyButton(string $text, string $url, $plainText = '');
 	 * Adds a logo and a text to the footer. <br> in the text will be replaced by new lines in the plain text email
 	 *
 	 * @param string $text If the text is empty the default "Name - Slogan<br>This is an automatically sent email" will be used
+	 * @param string $lang Optional language to set the default footer in
 	 *
 	 * @since 12.0.0
 	 */
-	public function addFooter(string $text = '');
+	public function addFooter(string $text = '', string $lang = null);
```
Comment (nickvergessen) [inline lib/public/Mail/IEMailTemplate.php:147]:
```suggestion
	public function addFooter(string $text = '', ?string $lang = null);
```

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by kesselb** [comment]
https://help.nextcloud.com/t/error-declaration-of-oca-theming-themingdefaults-getslogan/79514

As ThemingDefaults extend OC_Defaults changing the method signature requires more work.

**Pull request merged**